src/text/raw_text_writer.rs

```diff
@@ -112,16 +120,47 @@ impl<W: Write> RawTextWriter<W> {
             && chars.all(|c| c == '$' || c == '_' || c.is_ascii_alphanumeric())
     }
 
+    /// Returns `true` if the provided text is an Ion keyword. Keywords like `true` or `null`
+    /// resemble identifiers, but writers must wrap them in quotes when using them as symbol text.
+    fn token_is_keyword(token: &str) -> bool {
+        const KEYWORDS: &[&str] = &["true", "false", "nan", "null"];
+        KEYWORDS.contains(&token)
+    }
```

Comment:
In this case it actually is trivial. Check out [the optimized output in the compiler explorer](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,selection:(endColumn:9,endLineNumber:2,positionColumn:9,positionLineNumber:2,selectionStartColumn:9,selectionStartLineNumber:2,startColumn:9,startLineNumber:2),source:'//+Type+your+code+here,+or+load+an+example.%0A++++pub+fn+token_is_keyword(token:+%26str)+-%3E+bool+%7B%0A++++++++const+KEYWORDS:+%26%5B%26str%5D+%3D+%26%5B%22true%22,+%22false%22,+%22nan%22,+%22null%22%5D%3B%0A++++++++KEYWORDS.contains(%26token)%0A++++%7D%0A%0A//+If+you+use+%60main()%60,+declare+it+as+%60pub%60+to+see+it+in+the+output:%0A//+pub+fn+main()+%7B+...+%7D%0A'),l:'5',n:'0',o:'Rust+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:r1610,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,libs:!(),options:'-C++++++++++++++++opt-level%3D3',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1,tree:'1'),l:'5',n:'0',o:'rustc+1.61.0+(Rust,+Editor+%231,+Compiler+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4). The function is inlined and the loop is unrolled, which is possible because the array is `const`. It boils down to a few `cmp`/`je`s:

```
example::token_is_keyword:
        cmp     rsi, 3
        je      .LBB0_7
        cmp     rsi, 5
        je      .LBB0_6
        cmp     rsi, 4
        jne     .LBB0_5
        mov     al, 1
        cmp     dword ptr [rdi], 1702195828
        je      .LBB0_9
        cmp     dword ptr [rdi], 1819047278
        jne     .LBB0_5
```

In any case, I haven't done any performance work on the text writer. Once we start profiling, I'm sure there will be plenty of lower-hanging fruit to choose from. :stuck_out_tongue: